### DISCUSSION TITLE
Renaming the "Blog" slug?

### DISCUSSION THREAD

**Discussion by jacksbuilds** [mkdocs-publisher/mkdocs-publisher#78] [Q&A] [answered]
Hi there!

First and foremost, I want to thank you for your work on this project. It fits my needs almost perfectly, and I'm very excited to start publishing publicly soon. If you have any way for us to support you (e.g. Ko-Fi or similar), I'd love to do so.

One minor hiccup I'm noticing: it seems that my blog has an odd slug to it:

![image](https://github.com/mkusz/mkdocs-publisher/assets/24984165/57485d70-2a9d-4d95-8710-2b56bb9c1d48)

- Expected: `http://127.0.0.1:8000/blog`
- Actual: `http://127.0.0.1:8000/index-0/`

I have the blog dir configured in my `mkdocs.yml` as shown below. If I try to name the file `index.md`, I get an odd error about the index being out of range, so I've settled on `README.md`:

![image](https://github.com/mkusz/mkdocs-publisher/assets/24984165/13ca6d72-be9e-435e-924d-ec0973939490)

My `README.md` at `99_blog\README.md` looks like this:

```
---
title: Blog
slug: blog
date: 2024-06-06 21:14:12
update: 2024-06-06 23:27:17
---
```

This seems to be the case with the official docs website, too:

![image](https://github.com/mkusz/mkdocs-publisher/assets/24984165/3d6e7bba-58b1-4834-8016-e72ec86225c6)

Also noteworthy that posts have the right URL slug:

![image](https://github.com/mkusz/mkdocs-publisher/assets/24984165/00c7163a-9737-4718-a36d-66ba3f6178c9)

My question is, can we customize that `index-0` slug somehow?

Many thanks!!! I appreciate your project a ton. Have a great day!

**Comment by mkusz** [comment]
Hi @jdoescyber,

First, I'm really happy that you find this project useful.

As for support, I will definitely create a proper way to support this project and for sure let everybody know on the project blog, so thanks in advance for that.

As for your finding, this is an obvious bug. I have created an #79 for that, but for the fix you will have to wait a moment (sorry for that). I have take a look at the possible solution for that but there is no easy fix at the moment. I need to finish release 1.4.0 since the code for it is complete (working on documentation for it). In this release, there will be some missing features for the meta plugin related to adding support for external links in menu (supported in vanilla mkdocs) and also possibility for redirections (right now there is an external plugin for that, but it's not compatible with the meta plugin, since they both build navigation in a totally different way). After release 1.4.0 I will work on the blog plugin since this is the oldest one in this project and the code needs a huge refactor. I will fix your issue while working on the refactor (there are many more things that I didn't take into consideration while this plugin was written at the beginning).

PS. I'm considering creating an Insiders program like one in [MkDocs Material](https://squidfunk.github.io/mkdocs-material/insiders/). Will you be interested in participating?

PS2. If you have any future requests, I will be glad to hear about it and extend this project even further.

**Comment by mkusz** [comment, marked as answer]
Hi @jdoescyber,
In the upcoming v1.4.2 this will be fixed. To change slug, you have to change `slug` key value in `mkdocs.yaml" file (have a look at this [documentation part](https://mkdocs-publisher.github.io/setup/general/pub-blog/#general). In general, this will be changed in v2.0.0 to allow changing this setting over `README.md` file, so it will be unified with pub-meta plugin.
